samples/core/tutorials/estimators/linear.ipynb

```diff
@@ -1399,6 +1422,211 @@
+        "  weight_values = [model.get_variable_value(name) for name in weight_names]\n",
+        "\n",
+        "  weights_flat = np.concatenate([item.flatten() for item in weight_values], axis=0)\n",
+        "\n",
+        "  return weights_flat\n",
+        "\n",
+        "weights_flat = get_flat_weights(model)\n",
+        "weights_flat_l1 = get_flat_weights(model_l1)\n",
+        "weights_flat_l2 = get_flat_weights(model_l2)"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "GskJmtfmL0p-",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "The models have many zero-vlaued weights  caused by unused hash bins (There are many more hash bins than categories in some columns). We will mask these weights when viewing the weight distributions:"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "rM3agZe3MT3D",
+        "colab_type": "code",
+        "colab": {
+          "autoexec": {
+            "startup": false,
+            "wait_interval": 0
+          }
+        }
+      },
+      "cell_type": "code",
+      "source": [
+        "weight_mask = weights_flat != 0\n",
+        "\n",
+        "weights_base = weights_flat[weight_mask]\n",
+        "weights_l1 = weights_flat_l1[weight_mask]\n",
+        "weights_l2 = weights_flat_l2[weight_mask]"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "NqBpxLLQNEBE",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "Now plot the distributions:"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "IdFK7wWa5_0K",
+        "colab_type": "code",
+        "colab": {
+          "autoexec": {
+            "startup": false,
+            "wait_interval": 0
+          }
+        }
+      },
+      "cell_type": "code",
+      "source": [
+        "plt.figure()\n",
+        "_ = plt.hist(weights_base, bins=np.linspace(-3,3,30))\n",
+        "plt.title('Base Model')\n",
+        "plt.ylim([0,500])\n",
+        "\n",
+        "plt.figure()\n",
+        "_ = plt.hist(weights_l1, bins=np.linspace(-3,3,30))\n",
+        "plt.title('L1 - Regularization')\n",
+        "plt.ylim([0,500])\n",
+        "\n",
+        "plt.figure()\n",
+        "_ = plt.hist(weights_l2, bins=np.linspace(-3,3,30))\n",
+        "plt.title('L2 - Regularization')\n",
+        "_=plt.ylim([0,500])\n",
+        "\n"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "Mv6knhFa5-iJ",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "Both types of regularization squeeze the distribution of weights towards zero. L2 regularization has a greater effect in the tails of the distribution eliminating extreme weights. L1 regularization produces more exactly-zero values (In this case it sets ~200 to zero)."
```

Comment:
Copy edit:

Both types of regularization squeeze the distribution of weights towards zero. L2 regularization has a greater effect in the tails of the distribution eliminating extreme weights. L1 regularization produces more exactly-zero values, in this case it sets ~200 to zero.

Reply:
Done